### PR TITLE
update django versions

### PR DESCRIPTION
**Pull request by annashamray**
Note:
I tried to update to Django 4.1 but the tests failed, it will require some additional time, effort and separate issue.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -30,9 +29,10 @@ zip_safe = False
 include_package_data = True
 packages = find:
 install_requires =
-    django>=2.2
+    django>=3.2
     djangorestframework>=3.11.0
     django-filter
+    coreapi
```
Comment (sergei-maertens) [inline setup.cfg:35]: I don't understand why this is needed - isn't that part of drf?

File: testapp/settings.py [hunk 2]
```diff
@@ -24,8 +24,7 @@ def get_db():
 
 DATABASES = {"default": get_db()}
 
-# TODO: update to BigAutoField and makemigrations once we drop 2.2 support
-DEFAULT_AUTO_FIELD = "django.db.models.AutoField"
+DEFAULT_AUTO_FIELD = "django.db.models.BigAutoField"
```
Comment (sergei-maertens) [inline testapp/settings.py:27]: this actually needs to be set in the  `AppConfig` - the models themselves when installed as part of OZ don't use this setting so you'll end up with the reverse migration again.

**Review by sergei-maertens — DISMISSED** [review]

File: testapp/settings.py (hunk 2 above)
Comment (annashamray) [reply testapp/settings.py:27]: @sergei-maertens But it's only a part of the test app. The lib itself doesn't have models and model-specific settings 
I don't want to include it into  `AppConfig` for this reason

File: setup.cfg (hunk 1 above)
Comment (annashamray) [reply setup.cfg:35]: It looks like DRF wants to drop its support in the future versions
But it still has assertion for coreapi 
https://github.com/encode/django-rest-framework/blob/4f7e9ed3bbb0fb3567884ef4e3d340bc2f77bc68/rest_framework/filters.py#L33

**Review by joeribekker — APPROVED** [review]

File: testapp/settings.py (hunk 2 above)
Comment (sergei-maertens) [reply testapp/settings.py:27]: yeah, my bad!

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**